Doc/library/os.path.rst

```diff
@@ -298,10 +298,10 @@ the :mod:`glob` module.)
 .. function:: join(path, *paths)
 
    Join one or more path components intelligently.  The return value is the
-   concatenation of *path* and any members of *\*paths* with exactly one
-   directory separator following each non-empty part except the last, meaning
-   that the result will only end in a separator if the last part is empty.  If
-   a component is an absolute path, all previous components are thrown away
+   concatenation of *path* and any non-empty members of *\*paths* separated
+   by the platform's directory separator character. If the last member of *\*paths*
+   is empty, the result will end in the platform's directory separator.
```

Comment:
@serhiy-storchaka Revisiting this, here's one alternative wording without introducing new terms:

```suggestion
   by the platform's directory separator character. If the last member of *\*paths*
   is empty, the result will usually end with the platform's directory separator.
   This does not happen in cases for certain values of *path*, such as ``'c:'``
   on Windows.
```

That would keep the idea of the original (but incorrect) note of `the result will only end in a separator if the last part is empty`. If dropping the note entirely is preferred, then I'll happily defer judgement here and in that case feel free to commit the below suggestion instead:

```suggestion
   by the platform's directory separator character. Empty members of *\*paths*
   are ignored except for the last path.
```